backend/tarsy/models/processing_context.py

```
@@ -0,0 +1,192 @@
+        return self.alert_data.copy()
+    
+    def get_runbook_content(self) -> str:
+        """Get downloaded runbook content."""
+        return self.runbook_content or ""
+    
+    def get_previous_stages_results(self) -> List[tuple[str, AgentExecutionResult]]:
+        """
+        Get completed stage results in execution order.
+        
+        Returns results as ordered list of (stage_name, result) tuples.
+        Dict preserves insertion order (Python 3.7+) so iteration order = execution order.
```

Ruff (0.12.2)

Line too long (90 > 88)

(E501)